nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/resolver/TestDocumentModelResolver.java

```diff
@@ -537,6 +541,28 @@ public void testSerialization() throws Exception {
         assertEquals(idRef, readResolver.getReference(doc));
     }
 
+    @Test
+    public void testUserWithNoAccessToDocument() {
+        String username = "foo";
+        DocumentModelResolver resolver = new DocumentModelResolver();
+        Map<String, String> params = new LinkedHashMap<>();
+        params.put(PARAM_STORE, STORE_ID_REF);
+        resolver.configure(params);
+
+        DocumentModel adminDoc = resolver.fetch(DocumentModel.class, idRef);
+
+        assertNotNull(adminDoc);
+
+        ClientLoginModule.getThreadLocalLogin().push(new UserPrincipal(username, Collections.emptyList(), false, false),
+                null, null);
+
+        DocumentModel doc = resolver.fetch(DocumentModel.class, idRef);
+
+        assertNull(doc);
+
```

Comment:
Don't use so many empty lines (I think we've mentioned this several times before)